Gradle quarkusBuild fails with java.util.ServiceConfigurationError: org.eclipse.microprofile.config.spi.ConfigSourceProvider: io.quarkus.test.common.http.TestHTTPConfigSourceProvider not a subtype

As often in life when you fix one issue another one pops up. So the initial error got fixed and running `quarkusBuild` works however when using `quarkusDev` another issue pops up:

22:01:04: Executing task 'quarkusDev'...

```
Starting Gradle Daemon...
Gradle Daemon started in 1 s 758 ms
> Task :compileJava UP-TO-DATE
> Task :processResources UP-TO-DATE
> Task :classes UP-TO-DATE
OpenJDK 64-Bit Server VM warning: forcing TieredStopAtLevel to full optimization because JVMCI is enabled
Listening for transport dt_socket at address: 5005
Exception in thread "main" java.lang.RuntimeException: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.RuntimeException: java.util.ServiceConfigurationError: org.eclipse.microprofile.config.spi.ConfigSourceProvider: io.quarkus.test.common.http.TestHTTPConfigSourceProvider not a subtype
	at io.quarkus.dev.DevModeMain.main(DevModeMain.java:51)
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.RuntimeException: java.util.ServiceConfigurationError: org.eclipse.microprofile.config.spi.ConfigSourceProvider: io.quarkus.test.common.http.TestHTTPConfigSourceProvider not a subtype
	at io.quarkus.dev.DevModeMain.start(DevModeMain.java:108)
	at io.quarkus.dev.DevModeMain.main(DevModeMain.java:46)
	Suppressed: java.lang.NullPointerException
		at io.quarkus.dev.DevModeMain.close(DevModeMain.java:115)
		at io.quarkus.dev.DevModeMain.main(DevModeMain.java:45)
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.RuntimeException: java.util.ServiceConfigurationError: org.eclipse.microprofile.config.spi.ConfigSourceProvider: io.quarkus.test.common.http.TestHTTPConfigSourceProvider not a subtype
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:134)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:84)
	at io.quarkus.dev.DevModeMain.start(DevModeMain.java:104)
	... 1 more
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: java.util.ServiceConfigurationError: org.eclipse.microprofile.config.spi.ConfigSourceProvider: io.quarkus.test.common.http.TestHTTPConfigSourceProvider not a subtype
	at io.quarkus.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:259)
	at io.quarkus.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:39)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:131)
	... 3 more
Caused by: java.lang.RuntimeException: java.util.ServiceConfigurationError: org.eclipse.microprofile.config.spi.ConfigSourceProvider: io.quarkus.test.common.http.TestHTTPConfigSourceProvider not a subtype
	at io.quarkus.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:82)
	at io.quarkus.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:236)
	... 5 more
Caused by: java.util.ServiceConfigurationError: org.eclipse.microprofile.config.spi.ConfigSourceProvider: io.quarkus.test.common.http.TestHTTPConfigSourceProvider not a subtype
	at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:588)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1236)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1264)
	at java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1299)
	at java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1384)
	at java.base/java.lang.Iterable.forEach(Iterable.java:74)
	at io.smallrye.config.SmallRyeConfigBuilder.discoverSources(SmallRyeConfigBuilder.java:79)
	at io.smallrye.config.SmallRyeConfigBuilder.build(SmallRyeConfigBuilder.java:175)
	at io.quarkus.runtime.generated.Config.<clinit>(Config.zig:18268)
	at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:310)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:398)
	at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:79)
	at io.quarkus.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:60)
	... 6 more
	Suppressed: java.lang.NoClassDefFoundError: Could not initialize class io.quarkus.runtime.generated.Config
		at java.base/java.lang.Class.forName0(Native Method)
		at java.base/java.lang.Class.forName(Class.java:398)
		at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:83)
		... 7 more
	Suppressed: java.lang.RuntimeException: Failed to recover after failed start
		at io.quarkus.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:81)
		... 6 more
	Caused by: java.lang.reflect.InvocationTargetException
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
		at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.base/java.lang.reflect.Method.invoke(Method.java:566)
		at io.quarkus.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:77)
		... 6 more
	Caused by: java.util.ServiceConfigurationError: org.eclipse.microprofile.config.spi.ConfigSourceProvider: io.quarkus.test.common.http.TestHTTPConfigSourceProvider not a subtype
		at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:588)
		at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1236)
		at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1264)
		at java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1299)
		at java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1384)
		at java.base/java.lang.Iterable.forEach(Iterable.java:74)
		at io.smallrye.config.SmallRyeConfigBuilder.discoverSources(SmallRyeConfigBuilder.java:79)
		at io.smallrye.config.SmallRyeConfigBuilder.build(SmallRyeConfigBuilder.java:175)
		at io.quarkus.runtime.configuration.QuarkusConfigFactory.getConfigFor(QuarkusConfigFactory.java:33)
		at io.smallrye.config.SmallRyeConfigProviderResolver.getConfig(SmallRyeConfigProviderResolver.java:86)
		at io.smallrye.config.SmallRyeConfigProviderResolver.getConfig(SmallRyeConfigProviderResolver.java:74)
		at org.eclipse.microprofile.config.ConfigProvider.getConfig(ConfigProvider.java:91)
		at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:49)
		at io.quarkus.runtime.logging.LoggingSetupRecorder.handleFailedStart(LoggingSetupRecorder.java:75)
		... 11 more
> Task :quarkusDev
> Task :prepareKotlinBuildScriptModel UP-TO-DATE

BUILD SUCCESSFUL in 14s
3 actionable tasks: 1 executed, 2 up-to-date
22:01:18: Task execution finished 'quarkusDev'.

```

_Originally posted by @SR-Lut3t1um in https://github.com/quarkusio/quarkus/issues/7234#issuecomment-587158415_

--- a/extensions/qute/deployment/pom.xml
+++ b/extensions/qute/deployment/pom.xml
@@ -33,6 +33,7 @@
         <dependency>
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-junit5-internal</artifactId>
+            <scope>test</scope>
         </dependency>
 
     </dependencies>

